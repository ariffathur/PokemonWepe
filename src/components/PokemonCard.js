import React from "react";
import { View, StyleSheet, Image, Text, TouchableOpacity } from "react-native";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";
import { getPokeIdFromURL } from "../transforms/common";

const PokemonCard = (props) => {
  const { item, index, navigation } = props;

  console.tron.log(item.url);

  return (
    <TouchableOpacity
      onPress={() =>
        navigation.navigate("DetailScreen", {
          pokemonDetailURL: item.url,
        })
      }
      style={[
        style.cardContainer,
        style.shadow,
        index % 2 === 0 ? style.cardOdd : style.cardEven,
      ]}
    >
      <Image
        style={style.image}
        resizeMode={"contain"}
        source={{
          uri: `https://pokeres.bastionbot.org/images/pokemon/${getPokeIdFromURL(
            item.url
          )}.png`,
        }}
      />
      <View style={style.descriptionContainer}>
        <Text style={style.title} numberOfLines={2}>
          {item.name}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

export default PokemonCard;

const style = StyleSheet.create({
  cardContainer: {
    width: wp("45%"),
    height: hp("25%"),
    backgroundColor: "#FFFF",
    borderRadius: 8,
  },
  descriptionContainer: {
    alignItems: "center",
  },
  title: { fontSize: 14, fontWeight: "bold", color: "#5D6A70" },
  image: {
    width: wp("45%"),
    height: hp("20%"),
  },
  shadow: {
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.2,
    shadowRadius: 1.41,

    elevation: 2,
  },
  cardOdd: {
    marginBottom: 12,
    marginRight: 8,
  },
  cardEven: {
    marginBottom: 12,
  },
});
