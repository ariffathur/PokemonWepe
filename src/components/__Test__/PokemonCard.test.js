import React from "react";
import renderer from "react-test-renderer";

import PokemonCard from "../PokemonCard";

describe("<PokemonCard />", () => {
  const item = {
    name: "bulbasaur",
    url: "https://pokeapi.co/api/v2/pokemon/2/",
  };
  it("has 2 child", () => {
    const tree = renderer.create(<PokemonCard item={item} />).toJSON();
    expect(tree.children.length).toBe(2);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<PokemonCard item={item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
