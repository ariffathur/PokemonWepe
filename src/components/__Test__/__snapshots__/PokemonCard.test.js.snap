// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonCard /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFF",
      "borderRadius": 8,
      "elevation": 2,
      "height": 333.5,
      "marginBottom": 12,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
      "width": 337.5,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://pokeres.bastionbot.org/images/pokemon/2.png",
      }
    }
    style={
      Object {
        "height": 267,
        "width": 337.5,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        Object {
          "color": "#5D6A70",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      bulbasaur
    </Text>
  </View>
</View>
`;
